tests/core/test_save_restore.py

```diff
@@ -85,12 +101,66 @@ def setup_validation_data(self, val_data_config: Union[DictConfig, Dict]):
     def setup_test_data(self, test_data_config: Union[DictConfig, Dict]):
         self._test_dl = None
 
+    @classmethod
     def list_available_models(cls):
         return []
 
 
+class MockModelWithChildren(MockModel):
+    """
+    Mock Model, can contain 2 children (other NeMo models)
+    """
+
+    def __init__(self, cfg, trainer=None):
+        super().__init__(cfg=cfg, trainer=trainer)
+
+        # child 1
+        if cfg.get('child1_model_path') is None and cfg.get('child1_model') is None:
+            self.child1_model = None
+        else:
+            # if `child1_model_path` is set, model should be restored from nemo checkpoint
+            # otherwise model is constructed from config in `child1_model`
+            if cfg.get('child1_model_path') is not None:
+                self.child1_model = ModelPT.restore_from(cfg.child1_model_path)
+                # path in config should be set to `None` to restore model from config after saving
+                self.cfg.child1_model_path = None
+                # config for child model should be stored in base model config
+                self.cfg.child1_model = self.child1_model.cfg
```

Comment:
Is this for demonstration or do we need to ask model devs to do this? If so, can we automate it somehow.

Reply:
If the model is loaded from checkpoint (see a comment about 2 ways to construct child models), the developer should assign a config.

I also thought about doing this automatically (since we can use `__getattr__` on `ModelPT` to do this), but:
- there are cases when the model will be directly constructed from config, so we don't need to assign it once more
- this will break existing code: there is existing NLP code, that uses additional NeMo models as attributes, but doesn't save them (actually I catched errors when tried to do everything recursively)
- if the model is not needed to be saved (assigned as attribute, but then deleted), the developer will have to delete `cfg.child1_model` directly after 'magic' construction of this attribute.

So, I think it's better to ask developers to do it manually if constructing child models from `.nemo` checkpoints.

Reply:
Let's not modify getattr. Very annoying to do it correctly and brittle for future. 

However it is also very easy to forget this step and completely break your implementation. 

Add a test that checks this - a model which is nested but "forgets" to set the config. Raise an error. 

What about doing this - inside of save_to, check if any module is ModelPT and if so, check the config is available other wise raise error

Reply:
Or do this - inside save to check if the config is set, if yes don't do anything else otherwise use self.nested_module_i.cfg and set the config automatically inside of self.cfg